node/subsystem-types/src/runtime_client.rs

```diff
@@ -0,0 +1,449 @@
+	///
+	/// NOTE: This function is only available since parachain host version 2.
+	async fn pvfs_require_precheck(&self, at: Hash) -> Result<Vec<ValidationCodeHash>, ApiError>;
+
+	/// Fetch the hash of the validation code used by a para, making the given `OccupiedCoreAssumption`.
+	///
+	/// NOTE: This function is only available since parachain host version 2.
+	async fn validation_code_hash(
+		&self,
+		at: Hash,
+		para_id: Id,
+		assumption: OccupiedCoreAssumption,
+	) -> Result<Option<ValidationCodeHash>, ApiError>;
+
+	/// Returns all onchain disputes.
+	/// This is a staging method! Do not use on production runtimes!
+	async fn staging_get_disputes(
+		&self,
+		at: Hash,
+	) -> Result<Vec<(SessionIndex, CandidateHash, DisputeState<BlockNumber>)>, ApiError>;
+
+	// === BABE API ===
+
+	/// Return the genesis configuration for BABE. The configuration is only read on genesis.
+	async fn configuration(&self, at: Hash) -> Result<BabeGenesisConfiguration, ApiError>;
+
+	/// Returns the slot that started the current epoch.
+	async fn current_epoch_start(&self, at: Hash) -> Result<Slot, ApiError>;
+
+	/// Returns information regarding the current epoch.
+	async fn current_epoch(&self, at: Hash) -> Result<Epoch, ApiError>;
+
+	/// Returns information regarding the next epoch (which was already
+	/// previously announced).
+	async fn next_epoch(&self, at: Hash) -> Result<Epoch, ApiError>;
+
+	/// Generates a proof of key ownership for the given authority in the
+	/// current epoch. An example usage of this module is coupled with the
+	/// session historical module to prove that a given authority key is
+	/// tied to a given staking identity during a specific session. Proofs
+	/// of key ownership are necessary for submitting equivocation reports.
+	/// NOTE: even though the API takes a `slot` as parameter the current
+	/// implementations ignores this parameter and instead relies on this
+	/// method being called at the correct block height, i.e. any point at
+	/// which the epoch for the given slot is live on-chain. Future
+	/// implementations will instead use indexed data through an offchain
+	/// worker, not requiring older states to be available.
+	async fn generate_key_ownership_proof(
+		&self,
+		at: Hash,
+		slot: Slot,
+		authority_id: AuthorityId,
+	) -> Result<Option<OpaqueKeyOwnershipProof>, ApiError>;
+
+	/// Submits an unsigned extrinsic to report an equivocation. The caller
+	/// must provide the equivocation proof and a key ownership proof
+	/// (should be obtained using `generate_key_ownership_proof`). The
+	/// extrinsic will be unsigned and should only be accepted for local
+	/// authorship (not to be broadcast to the network). This method returns
+	/// `None` when creation of the extrinsic fails, e.g. if equivocation
+	/// reporting is disabled for the given runtime (i.e. this method is
+	/// hardcoded to return `None`). Only useful in an offchain context.
+	async fn submit_report_equivocation_unsigned_extrinsic(
```

Comment:
Do we really call all of these Babe functions?

Reply:
Not all were needed, removed unused ones 👍